include/eve/traits/overload/default_behaviors.hpp

```diff
@@ -28,18 +29,28 @@ namespace eve
   {
     using base = decorated_with<OptionsValues, Options...>;
 
-    template<typename T>
-    constexpr auto operator[](T t) const
-    requires( requires(base const& b) { b[t];} )
+    template<callable_options O> constexpr auto operator[](O const& opts) const
+    {
+      return  Func<O>{opts};
+    }
+
+    template<typename T> constexpr auto operator[](T t) const requires( requires(base const& b) { b[t];} )
     {
       auto new_traits = base::operator[](t);
       return  Func<decltype(new_traits)>{new_traits};
     }
 
-    template<typename T> void operator[](T t) const =delete;
+    // TEMPORARY - Map old decorator to the new ones
+    template<typename T> auto operator[](T const& t) const requires(decorator<T>)
+    {
+      return (*this)[as_option(t)];
+    }
+
+    template<typename T> void operator[](T t) const
+    // This requires is also TEMPORARY
+    requires( !callable_options<T> && !requires(base const& b) { b[t];} && !decorator<T>) =delete;
```

Comment:
Why do you need this requires?

Reply:
I needed it at one poitn when I tried to support old options. Maybe it is not needed anymore, will check and report